wizard_apps.ipynb

```diff
@@ -0,0 +1,325 @@
+    "        \"Format and show the pizza configuration.\"\n",
+    "        if change[\"new\"]:\n",
+    "            self.configuration_label.value = f\"<h4>Configuration</h4><pre>{json.dumps(change['new'], indent=2)}</pre>\"\n",
+    "        else:\n",
+    "            self.configuration_label.value = (\n",
+    "                \"<h4>Configuration</h4>[Please configure your pizza]\"\n",
+    "            )\n",
+    "\n",
+    "    def submit_order(self, button):\n",
+    "        \"Submit the order and simulate the delivery.\"\n",
+    "        self.order = OrderTracker(configuration=self.configuration)\n",
+    "\n",
+    "    def _update_state(self, _=None):\n",
+    "        \"Update the step's state based on the order status and configuration traits.\"\n",
+    "        if self.order:  # the order has been submitted\n",
+    "            self.state = self.State.SUCCESS\n",
+    "        elif self.configuration:  # the order can be submitted\n",
+    "            self.state = self.State.CONFIGURED\n",
+    "        else:\n",
+    "            self.state = self.State.INIT\n",
+    "\n",
+    "    @traitlets.observe(\"state\")\n",
+    "    def _observe_state(self, change):\n",
+    "        \"\"\"Enable the order button once the step is in the \"configured\" state.\"\"\"\n",
+    "        self.order_button.disabled = change[\"new\"] != self.State.CONFIGURED\n",
+    "\n",
+    "\n",
+    "class TrackOrderStep(ipw.VBox, WizardAppWidgetStep):\n",
+    "\n",
+    "    # We receive the order from the previous step and then display information about\n",
+    "    # its state in this widget.\n",
+    "    order = traitlets.Instance(OrderTracker, allow_none=True)\n",
+    "\n",
+    "    def __init__(self, **kwargs):\n",
+    "        # We will represent the current order status with this dedicated widget.\n",
+    "        # It is typically a good idea to strictly separate application logic and\n",
+    "        # its representation within the user interface.\n",
+    "        self.status_widget = OrderProgressWidget()\n",
+    "\n",
+    "        super().__init__([self.status_widget], **kwargs)\n",
+    "\n",
+    "    @traitlets.observe(\"order\")\n",
+    "    def _observe_order(self, change):\n",
+    "        \"Couple a newly received order to the status output widget.\"\n",
+    "        # Note: Technically we must unlink old orders, but this is ommitted for brevity.\n",
+    "\n",
+    "        if change[\"new\"]:\n",
+    "            # Order received, immediate state change:\n",
+    "            self.state = self.State.ACTIVE\n",
+    "\n",
+    "            # Now follow the order status for subsequent state changes:\n",
+    "            change[\"new\"].observe(self._update_state, [\"status\"])\n",
+    "\n",
+    "            # Link the order status to the widget displaying it to the user:\n",
+    "            ipw.dlink((change[\"new\"], \"status\"), (self.status_widget, \"status\"))\n",
+    "\n",
```

Comment:
```suggestion
    "        else:\n",
    "            self.state = self.State.INIT\n",
    "\n",
```

Otherwise the third step fails to reset.

Reply:
Thx a lot! 😄